### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by AmaliaGM**
## Project requirements

- Watch the video with the UI example again (you have seen it while reading the Sneak Peek).
UI example
- Your console app should behave in the same way as it is in the example.
- Create a app.rb file that will serve as your console app entry-point. It should have methods that do the following:
List all books.
List all people.
Create a person (teacher or student, not a plain Person).
Create a book.
Create a rental.
List all rentals for a given person id.
- In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
Present the user with a list of options to perform.
Lets users choose an option.
If needed, ask for parameters for the option.
Have a way to quit the app.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require_relative './app'
```
Comment (rustam-tolipov) [inline main.rb]: - [ ] You did great job. However, If I run the app I'm getting several errors. Kindly fix them to make your project approve. In order to check your project, your app need to be fully functional and not showing errors. You can see errors in this video:

https://user-images.githubusercontent.com/69011963/198720498-e78895e9-bf7f-4fe6-8aa5-76f46b231a32.mov

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @AmaliaGM, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @AmaliaGM 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights 🎯 
✔️ Descriptive PR
✔️ Used GitFlow correctly
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] An error that didn't exist previously now occurs in the `student.rb` file when I try to start the app. Please see the image below and try to fix it 👇 
image


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -12,26 +13,84 @@ def initialize(age, name, id, parent_permission: true)
...
-capitalized_person = CapitalizeDecorator.new(person)
-capitalized_person.correct_name
-capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
-capitalized_trimmed_person.correct_name
+  def self.create_person(persons)
+    print 'Do you want to create a student (1) or teacher (2) [Input a number]: '
+    option = gets.chomp.to_i
+    create_student.call
+    create_teacher.call
+    case option
```
Comment (AmaduKamara) [inline person.rb]: You're doing great so far, but just as the previous reviewer highlighted, the creation of a person is still giving errors and crashes the program. 

- [ ] When I'm creating a person, a student, or a teacher, the error comes from this section of your code. Kindly check this method and fix the error given when selecting either Y/N for parent permission for every creation of a person.

**See the screenshot below**
image

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi @AmaliaGM,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍 
✅ Linter checks passed 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all retals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def options(input)
+    case input
+    when '1' then list_all_books
+    when '2' then list_all_persons
```
Comment (Ismailco) [inline app.rb]: - [ ] As the previous reviewers mentioned your app produces some errors, kindly fix them so your app is working properly
- When listing the people the app crashes and produces this error:
Screenshot_20221029_022441

- When trying to create a rental the app crashes and produces this error:
Screenshot_20221029_022528

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @AmaliaGM, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 4]
```diff
@@ -2,36 +2,94 @@
...
   end
 
-  def correct_name
-    @name
+  def self.create_student
+    print 'Classroom: '
+    classroom = gets.chomp
+
+    print 'Age: '
+    age = Interger(gets.chomp)
```
Comment (victorokoroji) [inline person.rb]: - [ ] I got this error while trying to create a student. It seems you misspelled Integer 👇🏽 

image

File: person.rb [hunk 5]
```diff
@@ -2,36 +2,94 @@
...
+    print 'Has parent permission? [Y/N]: '
+    parent_permission_user = gets.chomp
+    case parent_permission_user.downcase
+    when 'y'
+      true
+    else
+      false
+    end
+
+    person = Student.new(classroom, age, name, parent_permission)
```
Comment (victorokoroji) [inline person.rb]: - [ ] After correcting the name error locally, i still cant create/add a STUDENT. I was getting this error. Kindly look into it 

image

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @AmaliaGM, 

Great work on making the changes requested by a previous reviewer 👏 🔥 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Top highlight 🎯

- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @AmaliaGM,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**